src/main/scala/com.snowplowanalytics.snowplow.micro/SqliteStorage.scala

```
@@ -200,62 +206,123 @@ private[micro] class SqliteStorage(xa: Transactor[IO], maxEvents: Option[Int]) e
+      }
+    }.map(_._1)
+  }
 
   /** Create SQLite storage with file-based database. */
   def file(dbPath: String, maxEvents: Option[Int]): Resource[IO, SqliteStorage] = {
-    val url = s"jdbc:sqlite:$dbPath?journal_mode=WAL"
-    create(url, maxEvents)
-  }
-
-  /** Create SQLite storage with in-memory database.
-    * For tests only (does not support multiple connections). */
-  def inMemory(maxEvents: Option[Int]): Resource[IO, SqliteStorage] = {
-    val url = "jdbc:sqlite::memory:"
+    val params = List(
+      "journal_mode=WAL",   // Write-Ahead Logging for better concurrency
+      "busy_timeout=30000", // Wait up to 30 seconds instead of immediate BUSY error
+      "synchronous=NORMAL", // Balance between safety and performance
+      "cache_size=-20000",  // 20MB cache (negative = KB)
+      "temp_store=MEMORY"   // Use memory for temporary tables
+    )
+    val url = s"jdbc:sqlite:$dbPath?${params.mkString("&")}"
     create(url, maxEvents)
   }
 
   private def create(url: String, maxEvents: Option[Int]): Resource[IO, SqliteStorage] = {
     for {
+      ec <- databaseExecutionContext
       xa <- HikariTransactor.newHikariTransactor[IO](
         driverClassName = "org.sqlite.JDBC",
         url = url,
         user = "", // SQLite doesn't use user/password
         pass = "",
-        connectEC = databaseExecutionContext
+        connectEC = ec
       )
+      _ <- Resource.eval(IO {
+        xa.kernel match {
+          case ds: com.zaxxer.hikari.HikariDataSource =>
+            val config = ds.getHikariConfigMXBean
+            config.setMaximumPoolSize(5)       // Allow multiple concurrent readers (WAL mode supports this)
+            config.setMinimumIdle(1)           // Keep one connection alive
+            config.setConnectionTimeout(30000) // 30 second connection timeout
+            config.setIdleTimeout(600000)      // 10 minute idle timeout
+          case _ =>
+            ()
+        }
+      }.handleErrorWith { err =>
```

I'm not sure your intention on this line, but `handleErrorWith` is probably not want you want.  It catches and ignores the error.  You probably don't want to ignore this error and carry on anyway?